asyncapi/asgardeo/asyncapi.yml

```diff
@@ -230,6 +234,52 @@ components:
         serviceProvider:
           type: string
           example: My Account
+        authSteps:
+          type: array
+          description: A list of authentication steps
+          items:
+            $ref: '#/components/schemas/authStep'
+    LoginFailedEvent:
+      type: object
+      properties:
+        securityData:
+          $ref: '#/components/schemas/GenericSecurityData'
+        eventData:
+          $ref: '#/components/schemas/LoginFailedData'
+    LoginFailedData:
+      type: object
+      properties:
+        ref:
+          type: string
+          example: >-
+            https://api.asg.io/t/myorg/scim2/Users/36a4541a-1055-4986-872c-cdf2faa7a468
+        organizationId:
+          type: int
+          example: '1234'
+        organizationName:
+          type: string
+          example: myorg
+        userId:
```

Comment:
As a suggestion, it is better to provide what fields are mandatory in the AsyncAPI specification itself. This helps when matching them to Ballerina records. By checking the schema, you can see which Ballerina record fields could be optional. Also, it's helpful to mark nullable fields, making it easier to mark Ballerina record fields as 'nillable'.